ttnn/cpp/ttnn/tensor/types.cpp

```diff
@@ -293,3 +293,24 @@ MemoryConfig load_memory_config(const std::string& file_name) {
 }  // namespace tt_metal
 
 }  // namespace tt
+
+namespace ttnn {
+namespace types {
+
+const uint32_t Shape::operator[](std::int64_t index) const { return this->value.without_padding()[index]; }
+
+const uint32_t* Shape::begin() const {return this->value.without_padding().begin(); }
+const uint32_t* Shape::end() const {return this-> value.without_padding().end(); }
+
+tt::tt_metal::Array4D Shape::to_array_4D() const {
```

Comment:
If it's limited usage and it is possible to convert this to a utility method instead of a method on the class, I'd prefer that. Then we can just have a generic version that is templated on rank.

Reply:
Should I just remove it? I think we don't really this either.

Reply:
yeah, let's remove it